.github/workflows/test-and-lint-js.yml

```diff
@@ -0,0 +1,42 @@
+name: Test & lint JS codebase
+
+on:
+  pull_request:
+    branches: [ "main" ]
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths:
+      - '**.js'
+      - '**.ts'
+      - '**.vue'
+      - 'package.json'
+      - 'package.lock'
+
+jobs:
+  test-and-lint-js:
+    name: Test & lint JS codebase
+    timeout-minutes: 10
+    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4.1.1 #https://github.com/actions/checkout
+
+      - name: Cache dependencies
+        uses: actions/cache@v3.3.2 #https://github.com/actions/cache
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-npm-dependencies-${{ hashFiles('package.lock') }}
+          restore-keys: ${{ runner.os }}-npm-dependencies
+
+      - name: Set up node
+        uses: actions/setup-node@v4 #https://github.com/actions/setup-node
+        with:
+          node-version: 20
+
+      - name: Install npm dependencies
+        working-directory: src
+        run: npm install
```

Comment:
I think `clean-install` will be better here, it wont update packages as `install` command does.

Please adjust it in another workflow too.

https://docs.npmjs.com/cli/v8/commands/npm-ci

Reply:
I do not have `package-lock.json` in `src` directory

Reply:
Ok, right, I didn't figure it out.